File: 0205-isomorphic-strings/0205-isomorphic-strings.cpp
class Solution {
public:
    bool isIsomorphic(string s, string t) {
        unordered_map<char, char> maps;
        unordered_map<char, char> mapt;
        if (s.size()!=t.size()){
            return false ;
        }
        for (int i= 0; i<s.size();i++){
            char char_s = s[i];

            char char_t = t[i];
             if (maps.find(char_s) == maps.end()) {
                maps[char_s] = char_t;
            }
            if (mapt.find(char_t) == mapt.end()) {
                mapt[char_t] = char_s;
            }

            if (maps[char_s] != char_t || mapt[char_t] != char_s) {
                return false;
        }
        
    }
    return true;}
};

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution(object):
    def isIsomorphic(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        if len(s) != len(t):
            return False

        maps = {}
        mapt = {}
        for i in range(len(s)):
            if s[i] not in maps:
                maps[s[i]] = t[i]
            if t[i] not in mapt:
                mapt[t[i]] = s[i]

            if maps[s[i]] != t[i] or mapt[t[i]] != s[i]:
                return False
        return True